src/js/spatial-navigation.js

```diff
@@ -0,0 +1,496 @@
+class SpatialNavigation {
+  /**
+   * Constructs a SpatialNavigation instance with initial settings.
+   * Initializes key codes for navigation, sets up the player instance,
+   * and prepares the spatial navigation system.
+   *
+   * @class
+   * @param {Object} player - The Video.js player instance to which the spatial navigation is attached.
+   * @param {Component|null} initialFocusedComponent - The component that should initially have focus
+   *                                                   when the spatial navigation system starts.
+   *                                                   If null or not provided, no component will be initially focused.
+   */
+  constructor(player, initialFocusedComponent) {
+    this.KEY_CODE = { 37: 'left', 38: 'up', 39: 'right', 40: 'down', 415: 'play', 19: 'pause', 417: 'ff', 412: 'rw' };
```

Comment:
Wouldn't it be possible to extend `keycode` with the missing keys to avoid having two ways of handling keys?

Let's say somewhere in the `utils` directory something like:
```javascript
import keycode from 'keycode';

keycode.codes.play = 415;
keycode.codes.ff = 417;
keycode.codes.rw = 412;

export keycode;
```

Reply:
Sure @amtins I can move the keycodes to another file.
@dzianis-dashkevich would be this a viable solution for you?

Reply:
I am fine with this

Reply:
Changed as per @amtins suggestion in the latest commit.